### PR TITLE
fix: cursor state after deleting on list or empty page

### PR DESCRIPTION
**Pull request by DiamondThree**

### CODE REVIEW

File: packages/blocks/src/__internal__/rich-text/rich-text-operations.ts [hunk 1]
```diff
@@ -159,11 +159,14 @@ export function handleLineStartBackspace(store: Store, model: ExtendedModel) {
       if (!parent || parent?.flavour === 'group') {
         const container = getContainerByModel(model);
         const previousSibling = getPreviousBlock(container, model.id);
-        if (previousSibling) {
+        if (previousSibling && previousSibling.flavour === 'paragraph') {
           store.captureSync();
+          let preTextLength = model.text?.length || 0;
           previousSibling.text?.join(model.text as Text);
           store.deleteBlock(model);
           asyncFocusRichText(store, previousSibling.id);
```
Comment (doodlewind) [inline packages/blocks/src/__internal__/rich-text/rich-text-operations.ts:167]: Can we switch current `asyncFocusRichText` implementation to your `quill.setSelection` approach, or just remove this line? We shouldn't set cursor twice with 2 different APIs.

**Review by doodlewind — CHANGES_REQUESTED** [review]
This PR looks a bit careless, the case is not too tricky to add test case alongside, and again hope you *reinstall* VSCode to avoid the basic lint issue :(

File: packages/blocks/src/__internal__/rich-text/rich-text-operations.ts (hunk 1 above)
Comment (doodlewind) [reply packages/blocks/src/__internal__/rich-text/rich-text-operations.ts:167]: See comment in #162

**Review by doodlewind — APPROVED** [review]

File: tests/paragraph.spec.ts [hunk 2]
```diff
@@ -310,3 +310,23 @@ test('The key "Up" respond when the cursor located in first paragraph  ', async
   await page.keyboard.press('ArrowUp', { delay: 50 });
   await assertPageTitleFocus(page);
 });
+
+
+test('After deleting a row of the text, the cursor will jump to the end of the previous row of the list', async ({ page }) => {
```
Comment (lawvs) [inline tests/paragraph.spec.ts:315]: You should use 'should' instead of 'will' in unit tests to indicate it is expected behavior.

Comment (doodlewind) [reply tests/paragraph.spec.ts:315]: Good catch!

**Pull request merged**